lib/device-log/ios-log.js

```diff
@@ -249,7 +252,25 @@ class IOSLog {
   async getAllLogs () {
     return this.logs;
   }
+
+  /**
+   * Get 'idevicesyslog' subprocess for a given udid.
+   * 
+   * This function caches and re-uses instances so that there's
+   * just one instance per device (cache by UDID)
+   */
+  static async _getIDeviceSysLogProcess (cmd, args, env, udid) {
+    if (!IOSLog.cachedIDeviceSysLogs[cmd]) {
```

Comment:
doesn't really matter, but we could get away with a 1-level object doing something like 

```js
IOSLog.cachedIDeviceSysLogs[`${cmd}_${udid}`]
```

Reply:
My first implementation only used the `udid`. I added the `cmd` key in-case there were different subprocesses running off the same `udid`. I can change it to only use `udid` if you don't think there's any issue with that.